src/Workspaces/Core/Portable/Editing/SyntaxGenerator.cs

```diff
@@ -1228,7 +1228,11 @@ public SyntaxNode AddMembers(SyntaxNode declaration, params SyntaxNode[] members
     /// <summary>
     /// Changes the <see cref="DeclarationModifiers"/> for the declaration.
     /// </summary>
-    public abstract SyntaxNode WithModifiers(SyntaxNode declaration, DeclarationModifiers modifiers);
+    public SyntaxNode WithModifiers(SyntaxNode declaration, DeclarationModifiers modifiers)
+        => WithModifiers<SyntaxNode>(declaration, modifiers);
+
+    internal abstract TSyntaxNode WithModifiers<TSyntaxNode>(TSyntaxNode declaration, DeclarationModifiers modifiers)
+        where TSyntaxNode : SyntaxNode;
```

Comment:
by having this method, we allow internal callers to be strongly typed, allowing them to have to avoid casting the return type back to the type of hte node they passed in.